### ISSUE TITLE
HTML entities in abstracts incorrectly parsed

### ISSUE THREAD

**Issue by ewohnlich** [biopython/biopython#1742]
### Setup

I am reporting a problem with Biopython version, Python version, and operating
system as follows:

2.7.9 (default, Dec 10 2014, 12:28:03) [MSC v.1500 64 bit (AMD64)]
CPython
Windows-7-6.1.7601-SP1
1.72

### Behaviour

In a recent change, Entrez started allowing HTML tags in some fields like title and abstract and Biopython was updated to support this. However, it seems that HTML entities, e.g. `&lt;`, that are returned by Entrez are being parsed by Biopython so that I would get it as e.g. "<". This is a problem because I would expect to be able to render the resulting as abstract as HTML, but this already rendered "<" can create rendering errors.

### Steps to reproduce

An example publication can be found here https://eutils.ncbi.nlm.nih.gov/entrez/eutils/efetch.fcgi?db=pubmed&id=28775130&retmode=xml. In the METHODS section of the abstract, they intend to underline a less than sign. In the XML code this appears to be correctly done as `"<u>&lt;</u>"`. However the parsed result I get back is `"<u><</u>"`. Browsers can struggle to read this and I have observed it simply not ending the underline tag.

**Comment by ewohnlich** [comment]
I would expect it to be the following (truncated)
`StringElement('Self-reported lifetime... (TSH &gt;4.5 mIU/L) compared with normal TSH (0.4-<u>&lt;</u>4.5 mIU/L) and for anti-TPO positivity. We also examined pesticide associations with TSH, T4 and T3 in multivariate linear regression models.', attributes={u'NlmCategory': u'METHODS', u'Label': u'METHODS'})`

**Comment by peterjc** [comment]
Your proposal does match the raw XML,

```
<AbstractText Label="METHODS" NlmCategory="METHODS">Self-reported ... hypothyroidism (TSH &gt;4.5 mIU/L) compared with normal TSH (0.4-<u>&lt;</u>4.5 mIU/L) and for anti-TPO positivity. We also examined pesticide associations with TSH, T4 and T3 in multivariate linear regression models.</AbstractText>
```

I wonder what historically plain less-than and greater-than signs were presented as when parser with Biopython? Would this be a change?

**Comment by mdehoon** [comment]
In the past, less-than and greater-than were returned as '<' and '>', not as '&lt;' and '&gt;'. This is actually done by the expat parser in the Python standard library (which is used by the Bio.Entrez parser).

The purpose of the parser is to convert XML to plain text (using the XML to organize the data). From this viewpoint, converting '&lt;' and '&gt;' to '<' and '>' makes sense; just the fact that some HTML tags are used does not mean that the Bio.Entrez parser should return valid HTML. Also consider that typically the output of the parser in Bio.Entrez is used as plain text, so then I think that '<' and '>' is more appropriate than '&lt;' and '&gt;' .

**Comment by ewohnlich** [comment]
@mdehoon I think you are getting caught by the same behavior that got me, where github is rendering what you are trying to paste as html entities. We need to put it in markdown as code (I've updated my original post)

**Comment by mdehoon** [comment]
@ewohnlich  Could you convert the text using the html module:
``` 
>>> import html
>>> text = "... compared with normal TSH (0.4-<u><</u>4.5 mIU/L) and ..."
>>> html.escape(text)
'... compared with normal TSH (0.4-&lt;u&gt;&lt;&lt;/u&gt;4.5 mIU/L) and ...'
```
I believe that would fix your problem.

**Comment by peterjc** [comment]
I don't think either of those are useful as the different meanings of the ``<`` have been lost (opening an XML tag, and a less than sign).

As @ewohnlich wrote earlier, I think  ``'... compared with normal TSH (0.4-<u>&gt;</u>4.5 mIU/L) and ...'`` be the most useful HTML fragment to aim for.

To me is seems pragmatic to just return the abstract text as is, with the HTML entries and escaping as performed by the NCBI. This would also work for MathML in #1801.

This would be a change from old behaviour where the user would expect escaped entries like ``&gt;`` to have been turned into ``<`` but that seems a price worth paying.

**Comment by mdehoon** [comment]
Sorry, I wasn't thinking clearly.
It's fairly easy to change less-than signs to `&lt;` (and all other HTML escaping needed) but it can be very confusing to users who expect plain text.
In the current example, `<u>&lt;</u>` is reasonable, but if the abstract contains plain text only, without any HTML tags, then it would be strange to have HTML-escaped text in it.
We could add an argument `escape` to `Entrez.read` and `Entrez.parse` that, if `True`, HTML-escapes all text. So if `escape` is `True`, then all strings returned by the parser are valid pieces of HTML; if `escape` is `False` (which is the default), then all strings returned by the parser are plain text.

**Comment by peterjc** [comment]
The proposed new argument would I think be a neat solution here 👍

**Comment by peterjc** [comment]
Fixed via #1811
